### PR TITLE
Thread safety in RawFileWriter + misc changes in Detectors/Raw

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

File: Detectors/Raw/include/DetectorsRaw/RawFileWriter.h [hunk 1]
```diff
@@ -49,28 +44,49 @@ class RawFileWriter
...
     //
     size_t nTFWritten = 0;    // number of TFs written
     size_t nRDHWritten = 0;   // number of RDHs written
     size_t nBytesWritten = 0; // number of bytes written
     //
     std::string fileName{};                // file name associated with this link
     std::vector<char> buffer;              // buffer to accumulate superpage data
-    const RawFileWriter* writer = nullptr; // pointer on the parent writer
+    RawFileWriter* writer = nullptr;       // pointer on the parent writer
+    std::mutex mtx;
```
Comment (sawenzel) [inline Detectors/Raw/include/DetectorsRaw/RawFileWriter.h]: are you using this mutex anywhere? You probably only need it if used across different function calls.

File: Detectors/Raw/src/RawFileWriter.cxx [hunk 2]
```diff
@@ -103,48 +129,71 @@ RawFileWriter::LinkData& RawFileWriter::registerLink(const RDH& rdh, const std::
...
 
+//===================================================================================
+
 //___________________________________________________________________________________
 void RawFileWriter::LinkData::addData(const IR& ir, const gsl::span<char> data, bool preformatted)
 {
   // add payload corresponding to IR
   LOG(DEBUG) << "Adding " << data.size() << " bytes in IR " << ir << " to " << describe();
-
+  std::lock_guard<std::mutex> lock(mtx);
```
Comment (davidrohr) [inline Detectors/Raw/src/RawFileWriter.cxx]: @sawenzel : it is used here:

File: Detectors/Raw/include/DetectorsRaw/SimpleRawReader.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#include <vector>
+#include <gsl/gsl>
+#include "Headers/RAWDataHeader.h"
+#include "DetectorsRaw/RawFileReader.h"
+
+namespace o2
+{
+
+namespace itsmft
```
Comment (aphecetche) [inline Detectors/Raw/include/DetectorsRaw/SimpleRawReader.h]: I guess that namespace should be changed ? (or the location of this file changed)

**Comment by shahor02** [comment]
@aphecetche right, thanks for spotting! Since this class is needed for my PR in preparation, will fix it there.

**Pull request merged**